test/ministro-contracts/ministroChain.js

```diff
@@ -99,40 +101,74 @@ function ProxyContract() {
     return results;
   };
 
-  app.updateRegistryAddress = async (registryAddress, txAttr, expectThrow) => {
-    const txAttrLocal = app.getTxAttr(txAttr);
-
-    const action = () => app.instance.updateRegistryAddress(registryAddress, txAttrLocal);
-
-    const results = await app.executeAction(action, txAttrLocal, 1, 'LogUpdateRegistryAddress', expectThrow);
-
-    if (!expectThrow) {
-      const logUpdateRegistryAddress = results.LogUpdateRegistryAddress[0];
-      assert.strictEqual(logUpdateRegistryAddress.newRegistryAddress.toLowerCase(), registryAddress, 'invalid newRegistryAddress');
-
-      const registry = await app.registryAddress();
-      assert.strictEqual(registry.toLowerCase(), registryAddress, 'registryAddress is not saved on blockchain');
-    }
+  async function trackEvent(contract, eventName, call) {
+    return new Promise((resolve, reject) => {
+      let eventResult;
+      let callResult;
+      let callExecuted = false;
+      let timeout;
+
+      function process() {
+        if (!callExecuted) {
+          // we must wait for the call
+          return;
+        }
+
+        if (!eventResult) {
+          // wait a bit
+          timeout = setTimeout(() => {
+            timeout = null;
+            process();
+          }, 10000);
```

Comment:
wait 10 seconds? can we wait 1 second? or even better 0.25s?

Reply:
It doesn't affect the performance at all. In case if no event is emitted we wait for 10 seconds until the test fails because "undefined" is returned as a result.

Reply:
@DZariusz can you move the function to minitsto tools?